src/drop_context.rs

```diff
@@ -128,3 +126,17 @@ where
         self.inner.call(Request::from_parts(head, body))
     }
 }
+
+impl<T, C> crate::client::Service for DropContextService<T, C>
+where
+    C: Send + 'static,
+    T: crate::client::Service<ReqBody = hyper::Body>,
+{
+    type ReqBody = ContextualPayload<hyper::Body, C>;
+    type Future = T::Future;
+
+    fn request(&self, request: Request<Self::ReqBody>) -> Self::Future {
+        let (head, body) = request.into_parts();
+        self.inner.request(Request::from_parts(head, body.inner))
+    }
+}
```

Comment:
Seem to be some missing tests?

Reply:
I'm not sure what you want to test?  The implementation is sound, otherwise it wouldn't compile.  I notice that there's no test for the server side implementation either....

Reply:
An example of how to construct it, so that when we update this, we satisfy the use case?

Reply:
:+1:

Reply:
Done (as a doctest for extra win)